src/components/DropdownList.tsx

```diff
@@ -0,0 +1,60 @@
+import React from 'react';
+import { Link } from 'react-router-dom';
+import styled from 'styled-components';
+
+export interface DropdownItem {
+	path: string;
+	title: string;
+	icon?: string;
+}
+interface Props {
+	items: DropdownItem[];
+}
+const DropdownMenu = styled.ul`
+	list-style: none;
+	text-align: start;
+	z-index: 2;
+	border: 1px solid rgba(0, 0, 0, 0.04);
+	box-shadow: 0 16px 24px 2px rgba(0, 0, 0, 0.14);
+	padding: 0;
+	margin: 0;
+	border-radius: 10px;
+`;
+const DropdownMenuItem = styled.li`
+	color: white;
+	cursor: pointer;
+	list-style: none;
+	&:hover {
+		background: #ededed;
+		list-style: none;
+		border-radius: 10px;
+	}
+`;
+const DropdownLink = styled(Link)`
+	display: block;
+	height: 100%;
+	text-decoration: none;
+	color: black;
+	padding: 1em;
+	&:hover {
+		color: #4961e1;
+	}
+`;
+function DropdownList(props: Props) {
```

Comment:
Is it possible to change this to `function DropdownList(items: DropdownItem[])` so the second interface isn't needed? (And below it'd just be `items.map`)
(I haven't tried this though, just a thought)

Reply:
I did think of that possibility and tried it but it broke my code in a number of spots and went against it (wanted to stick with what was already working). I think the way it is now may be better since the dropdown items can all be defined in one array (see ProfileDropdown.tsx `const DropdownItems`) so it's easier to input the values when using if that makes any sense

Reply:
All good! That completely makes sense :)

Reply:
Yeah react makes it so functions have one parameter, and contains keys to any data. It would be possible if items wasn't an array. You could change this to `function DropdownList({ items }: Props) { ... }` and then where you are doing `props.items` you can just say `items`